### ISSUE TITLE
Use case: using private mirrors of open-source packages

### ISSUE THREAD

**Issue by dmitris** [mattfarina/pkg#8]
Some companies mirror the open-source packages that are being used to ensure the source is available even if the upstream repository is no longer present and to avoid fetching from external sources for every build.  For this use case, it would be helpful to record the mirror repository (as done with the "repository" field in the proposal): for example "git@opensource.git.company.com:Masterminds/cookoo.git".  

For a vendoring tool that creates the metadata file, it would be great to be able to automatically map the original patterns such as https://go.googlesource.com or https://github.com/tchap/go-patricia/ to the mirrored repository location "git@opensource.git.company.com:golang/crypto.git" and "git@opensource.git.company.com:tchap/go-patricia".  This would make generating the specification file easier (without the need to go through the generating file and manually change all values in repository fields) and reduce the chance of inconsistencies creeping in.

**Comment by mattfarina** [comment]
You're absolutely right. On Glide we have a feature request open for this at Masterminds/glide#39.
